sources/PPU/PPU.cpp

```diff
@@ -424,4 +634,214 @@ namespace ComSquare::PPU
+				this->addToMainSubScreen(this->_backgrounds[BgName::bg3Priority]);
+			//sprites priority 1
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg2NoPriority]);
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg1NoPriority]);
+			//sprites priority 2
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg2Priority]);
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg1Priority]);
+			//sprites priority 3
+			if (this->_registers._bgmode.mode1Bg3PriorityBit)
+				this->addToMainSubScreen(this->_backgrounds[BgName::bg3Priority]);
+			break;
+		case 2:
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg2NoPriority]);
+			//sprites priority 0
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg1NoPriority]);
+			//sprites priority 1
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg2Priority]);
+			//sprites priority 2
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg1Priority]);
+			//sprites priority 3
+			break;
+		case 3:
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg2NoPriority]);
+			//sprites priority 0
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg1NoPriority]);
+			//sprites priority 1
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg2Priority]);
+			//sprites priority 2
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg1Priority]);
+			//sprites priority 3
+			break;
+		case 4:
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg2NoPriority]);
+			//sprites priority 0
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg1NoPriority]);
+			//sprites priority 1
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg2Priority]);
+			//sprites priority 2
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg1Priority]);
+			//sprites priority 3
+			break;
+		case 5:
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg2NoPriority]);
+			//sprites priority 0
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg1NoPriority]);
+			//sprites priority 1
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg2Priority]);
+			//sprites priority 2
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg1Priority]);
+			//sprites priority 3
+			break;
+		case 6:
+			//sprites priority 0
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg1NoPriority]);
+			//sprites priority 1
+			//sprites priority 2
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg1Priority]);
+			//sprites priority 3
+            break;
+		case 7:
+			// Not implemented
+		default:
+			break;
+		}
+	}
+
+	template <std::size_t DEST_SIZE, std::size_t SRC_SIZE>
+	void PPU::add_buffer(std::array<std::array<uint32_t, DEST_SIZE>, DEST_SIZE> &bufferDest, std::array<std::array<uint32_t, SRC_SIZE>, SRC_SIZE> &bufferSrc)
+	{
+		for (unsigned long i = 0; i < bufferSrc.size(); i++) {
+			for (unsigned long j = 0; j < bufferSrc[i].size(); j++) {
+				if (bufferSrc[i][j] > 0xFF) // 0xFF correspond to a black pixel with full brightness
```

Comment:
Is the sign correct? Shouldn't it be bufferSrc[i][j] <= 0xFF

Reply:
no why ?

Reply:
colors are RRGGBBAA